app/src/main/java/com/example/bottomnav/StructDecoder.java

```diff
@@ -6,42 +6,47 @@
 import java.util.Optional;
 
 public class StructDecoder implements DataDecoder {
-    private HashMap<String, DataDecoder> decodedPrimatives = new HashMap<>();
-    public ArrayList<VariableContents> variables;
-    private int size;
+    private ArrayList<Optional<DataDecoder>> decodedPrimatives;
 
-    public StructDecoder(ArrayList<VariableContents> con) {
-        variables = con;
-        size = variables.size();
-    }
 
-    @Override
-    public String valueToString() {
-        StringBuilder message = new StringBuilder();
-        for (int i = 0; i < variables.size(); i++) {
-            String varName = variables.get(i).name;
-            message.append(decodedPrimatives.get(varName).valueToString());
-            message.append("\n");
-        }
-        return message.toString();
+    public StructDecoder(ArrayList<Optional<DataDecoder>> decoded) {
+        decodedPrimatives = decoded;
     }
 
     @Override
-    public Object valueToRaw() {
-        return null;
+    public Optional<ArrayList> decodeToRaw(byte[] payload) {
```

Comment:
```suggestion
    public Optional<ArrayList<Optional>> decodeToRaw(byte[] payload) {
```

While you don't known the type parameter of the Optional you still do know that it is Optionals